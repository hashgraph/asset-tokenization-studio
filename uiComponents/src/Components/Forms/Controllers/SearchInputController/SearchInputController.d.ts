/// <reference types="react" />
import { type FieldValues } from "react-hook-form";
import type { InputProps } from "../../Input";
import type { InputControllerProps } from "../InputController/InputController";
export interface SearchInputControllerProps<T extends FieldValues> extends Omit<InputProps, "id" | "defaultValue" | "onClear">, InputControllerProps<T> {
    minSearchLength?: number;
    onSearch: (value: string) => void;
}
export declare const SearchInputController: <T extends FieldValues>({ variant, control, rules, id, defaultValue, minSearchLength, onSearch, ...props }: SearchInputControllerProps<T>) => JSX.Element;
