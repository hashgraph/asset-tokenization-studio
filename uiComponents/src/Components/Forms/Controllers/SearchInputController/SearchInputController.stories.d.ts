declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./SearchInputController").SearchInputControllerProps<import("react-hook-form").FieldValues>>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./SearchInputController").SearchInputControllerProps<import("react-hook-form").FieldValues>>;
export declare const MinInputSearchCustom: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./SearchInputController").SearchInputControllerProps<import("react-hook-form").FieldValues>>;
