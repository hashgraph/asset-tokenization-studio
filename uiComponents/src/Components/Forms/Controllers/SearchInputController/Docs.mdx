import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as SearchInputControllerStories from "./SearchInputController.stories";

<Meta of={SearchInputControllerStories} />

<Title />

<Primary />

## How to set up

SearchInputController is built with InputController

## Props

<ArgTypes of={SearchInputControllerStories} />

## Stories

<Stories title="" />
