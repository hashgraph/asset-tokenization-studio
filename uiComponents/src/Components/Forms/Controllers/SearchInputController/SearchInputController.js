var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { useController } from "react-hook-form";
import { InputController } from "../InputController/InputController";
import React from "react";
import { MagnifyingGlass } from "@phosphor-icons/react";
import { PhosphorIcon } from "@/Components/Foundations";
import { IconButton } from "@/Components/Interaction";
export var SearchInputController = function (_a) {
    var variant = _a.variant, control = _a.control, rules = _a.rules, id = _a.id, defaultValue = _a.defaultValue, _b = _a.minSearchLength, minSearchLength = _b === void 0 ? 3 : _b, onSearch = _a.onSearch, props = __rest(_a, ["variant", "control", "rules", "id", "defaultValue", "minSearchLength", "onSearch"]);
    var _c = useController({
        name: id,
        control: control,
        rules: rules,
        defaultValue: defaultValue
    }).field, onChangeDefault = _c.onChange, value = _c.value;
    var isDisabled = !value || value.length < minSearchLength;
    var handleSearch = function () {
        onSearch(value);
    };
    return (React.createElement(InputController, __assign({ id: id, control: control, variant: variant, isClearable: true, onChange: function (e) {
            onChangeDefault(e);
        }, onKeyDown: function (ev) {
            !isDisabled && ev.key === "Enter" && handleSearch();
        } }, props, { addonRight: React.createElement(IconButton, { "data-testid": "search-icon-button", size: "xs", "aria-label": "search", variant: "tertiary", icon: React.createElement(PhosphorIcon, { as: MagnifyingGlass }), isDisabled: isDisabled, onClick: handleSearch }) })));
};
