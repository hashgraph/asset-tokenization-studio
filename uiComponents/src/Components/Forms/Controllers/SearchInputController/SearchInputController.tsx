import { useController, type FieldValues } from "react-hook-form";
import type { InputProps } from "../../Input";
import type { InputControllerProps } from "../InputController/InputController";
import { InputController } from "../InputController/InputController";
import React from "react";
import { MagnifyingGlass } from "@phosphor-icons/react";
import { PhosphorIcon } from "@/Components/Foundations";
import { IconButton } from "@/Components/Interaction";

export interface SearchInputControllerProps<T extends FieldValues>
  extends Omit<InputProps, "id" | "defaultValue" | "onClear">,
    InputControllerProps<T> {
  minSearchLength?: number;
  onSearch: (value: string) => void;
}

export const SearchInputController = <T extends FieldValues>({
  variant,
  control,
  rules,
  id,
  defaultValue,
  minSearchLength = 3,
  onSearch,
  ...props
}: SearchInputControllerProps<T>) => {
  const {
    field: { onChange: onChangeDefault, value },
  } = useController<T>({
    name: id,
    control,
    rules,
    defaultValue,
  });

  const isDisabled = !value || value.length < minSearchLength;

  const handleSearch = () => {
    onSearch(value);
  };

  return (
    <InputController
      id={id}
      control={control}
      variant={variant}
      isClearable
      onChange={(e) => {
        onChangeDefault(e);
      }}
      onKeyDown={(ev) => {
        !isDisabled && ev.key === "Enter" && handleSearch();
      }}
      {...props}
      addonRight={
        <IconButton
          data-testid="search-icon-button"
          size="xs"
          aria-label="search"
          variant="tertiary"
          icon={<PhosphorIcon as={MagnifyingGlass} />}
          isDisabled={isDisabled}
          onClick={handleSearch}
        />
      }
    />
  );
};
