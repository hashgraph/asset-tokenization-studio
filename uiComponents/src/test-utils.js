import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
