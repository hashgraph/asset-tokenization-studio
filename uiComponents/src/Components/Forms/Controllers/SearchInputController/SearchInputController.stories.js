var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { SearchInputController } from "./SearchInputController";
import { useForm } from "react-hook-form";
var meta = {
    title: "Design System/Forms/Controllers/SearchInputController",
    component: SearchInputController,
    args: {
        id: "search",
        label: "Label",
        placeholder: "Placeholder",
        onSearch: function (value) { return console.log("Searching... ".concat(value)); }
    },
    argTypes: {},
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A22829&t=4DeUdPmWLzP51oPO-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) {
    var form = useForm({ mode: "onChange" });
    return React.createElement(SearchInputController, __assign({}, args, { control: form.control }));
};
export var Default = Template.bind({});
export var MinInputSearchCustom = Template.bind({});
MinInputSearchCustom.args = {
    placeholder: "Introduce 5 chars to search",
    minSearchLength: 5
};
