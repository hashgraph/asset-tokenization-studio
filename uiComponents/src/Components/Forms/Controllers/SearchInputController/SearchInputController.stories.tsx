import type { Meta, StoryFn } from "@storybook/react";
import React from "react";

import { SearchInputController } from "./SearchInputController";
import { useForm } from "react-hook-form";

const meta = {
  title: "Design System/Forms/Controllers/SearchInputController",
  component: SearchInputController,
  args: {
    id: "search",
    label: "Label",
    placeholder: "Placeholder",
    onSearch: (value) => console.log(`Searching... ${value}`),
  },
  argTypes: {},
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A22829&t=4DeUdPmWLzP51oPO-4",
    },
    docs: {},
  },
} as Meta<typeof SearchInputController>;
export default meta;

const Template: StoryFn<typeof SearchInputController> = (args) => {
  const form = useForm({ mode: "onChange" });
  return <SearchInputController {...args} control={form.control} />;
};

export const Default = Template.bind({});

export const MinInputSearchCustom = Template.bind({});
MinInputSearchCustom.args = {
  placeholder: "Introduce 5 chars to search",
  minSearchLength: 5,
};
