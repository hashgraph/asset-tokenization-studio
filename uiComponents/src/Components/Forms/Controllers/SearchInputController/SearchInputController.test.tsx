import { act, fireEvent } from "@testing-library/react";
import type { FieldValues } from "react-hook-form";
import { useForm } from "react-hook-form";
import type { SearchInputControllerProps } from "./SearchInputController";
import { SearchInputController } from "./SearchInputController";
import React from "react";
import { render } from "@/test-utils";

type DefaultProps = Omit<SearchInputControllerProps<FieldValues>, "control">;

describe(`<SearchInputController />`, () => {
  const defaultProps: DefaultProps = {
    id: "search",
    placeholder: "placeholder",
    label: "Search",
    onSearch: () => console.log("searching"),
  };

  const RenderWithForm = (props: DefaultProps) => {
    const localForm = useForm({
      mode: "onChange",
      criteriaMode: "all",
    });
    const { control } = localForm;

    return <SearchInputController control={control} {...props} />;
  };

  const factoryComponent = (props: DefaultProps = defaultProps) => {
    return render(<RenderWithForm {...props} />);
  };

  test("Component renders correctly", () => {
    const component = factoryComponent();

    const passwordField = component.getByTestId("search");

    const { placeholder } = defaultProps;
    expect(passwordField).toHaveProperty("placeholder", placeholder);

    expect(component.asFragment()).toMatchSnapshot("RenderWithPlaceholder");
  });

  test("should allow to search when min characters are reached", async () => {
    const component = factoryComponent();
    const searchButton = component.getByTestId("search-icon-button");
    const inputField = component.getByTestId("search");

    expect(searchButton).toBeDisabled();

    act(() => {
      fireEvent.change(inputField, { target: { value: "test" } });
    });

    expect(searchButton).toBeEnabled();
  });
});
