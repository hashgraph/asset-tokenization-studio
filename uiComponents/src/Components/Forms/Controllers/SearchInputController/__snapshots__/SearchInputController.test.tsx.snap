// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInputController /> Component renders correctly: RenderWithPlaceholder 1`] = `
<DocumentFragment>
  <div
    class="chakra-form-control css-1kxonj9"
    role="group"
  >
    <label
      class="css-mimi7r"
      for="search"
    >
      <span
        class="css-1e9661m"
        id="label"
      >
        Search
      </span>
      <div
        class="chakra-input__group css-pwmwmr"
        data-group="true"
      >
        <input
          class="chakra-input css-16qaat3"
          data-testid="search"
          id="search"
          name="search"
          placeholder="placeholder"
          value=""
        />
        <div
          class="chakra-input__right-element css-1akgnmh"
        >
          <button
            aria-label="search"
            class="chakra-button css-18vb5xg"
            data-testid="search-icon-button"
            disabled=""
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-kpbtv0"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M229.66,218.34l-50.07-50.06a88.11,88.11,0,1,0-11.31,11.31l50.06,50.07a8,8,0,0,0,11.32-11.32ZM40,112a72,72,0,1,1,72,72A72.08,72.08,0,0,1,40,112Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </label>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
